File: java/freeman/command/macro/GarageDoor.java
public class GarageDoor {
    boolean open=false;
    public GarageDoor(){}
    public void open() {
        if (open){
            System.out.println("Door is already opened");
        }
        else{
            open=true;
            System.out.println("Open the door");
        }
	
    }
    public void close() {
	if (open){
            open=false;
            System.out.println("Close the door!");
        }
        else{
            System.out.println("Door is already closed");
        }
    }
}

File: php/oop-study/lesson3/Undo/devices/GarageDoor.php
<?php
class GarageDoor{
	//Класс управления дверью гаража
	private $open=false;
	public function open() {//Открыть дверь
		if ($this->open){
			echo "Door is already opened<br>";
		}
		else{
			$this->open=true;
			echo "Open the door<br>";
		}		
	}
	public function close() {//Закрыть дверь
		if ($this->open){
			$this->open=false;
			echo "Close the door!<br>";
		}
		else{
			echo "Door is already closed<br>";
		}
	} 
} 
?>